h5netcdf/tests/test_h5netcdf.py

```diff
@@ -1191,8 +1187,21 @@ def test_nc4_non_coord(tmp_local_netcdf):
         assert f.dimensions["x"].size == 0
         assert f.dimensions["x"].isunlimited()
         assert f.dimensions["y"].size == 2
-        assert list(f.variables) == ["y", "test"]
-        assert list(f._h5group.keys()) == ["_nc4_non_coord_y", "test", "x", "y"]
```

Comment:
do you remember why we tested for `_nc4)non_coord_y`. I couldn't figure out a pattern between the two.

It seems like we shouldn't be so assertive in this case.

Reply:
`_nc4_non_coord_y` is the created variable "y", `y` is the created dimensions "y". 

We check for this, since these are two different things and we want to see if it is created correctly.